woocommerce/woocommerce
# Use SelectTree in Parent Category field

### Submission Review Guidelines:

- I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
- I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
- I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/). 
- Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:



Use SelectTree in Parent Category field and improve SelectTree's label by using BaseControl.

Closes #37633 .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Open the new product manager
2. Go to the "Categories" field
3. Press "Create new" (the create new category modal should open)
4. Check that the label style has changed and looks the same as other product editor's fields.
5. Create a new category and fill the parent category using only the mouse.
6. Save and check if it saved correctly
7. Create a new category and fill the parent category using only the keyboard by typing an existing category.
8. Save and check if it saved correctly
9. Create a new category and fill the parent category using only the keyboard using the up, down, home, end keys. Select one using space.
10. Save and check if it saved correctly
11. Perform any additional smoke tests you'd like.
